newIDE/app/src/AssetStore/AssetStoreContext.js

```diff
@@ -259,6 +287,25 @@ export const AssetStoreStateProvider = ({
     [fetchAssetsAndFilters, assetShortHeadersById, isLoading]
   );
 
+  // Randomize asset packs when number of asset packs and private asset packs change
```

Comment:
instead of randomizing with an effect, should you do it just below the `setPrivateAssetPacks(privateAssetPacks);` (line 259) ?
You should have both arrays ready and you can set the value here.

Also, is there a need to have multiple arrays (assetPacks, privateAssetPacks, ordering) or can we just have 1 array in the state `assetPacks` that contain the random ordering of all assetPacks (public & private) ?